modules/javafx.web/src/main/native/Source/WebCore/platform/graphics/java/PathJava.cpp

```diff
@@ -490,10 +491,58 @@ void Path::apply(const PathApplierFunction& function) const
     }
 }
 
-bool Path::strokeContains(StrokeStyleApplier*, const FloatPoint&) const
+bool Path::strokeContains(StrokeStyleApplier *applier, const FloatPoint& p) const
 {
-    notImplemented();
-    return false;
+    ASSERT(m_path);
+
+    GraphicsContext& gc = scratchContext();
+    gc.save();
+
+    // Stroke style is set to SolidStroke if the path is not dashed, else it
+    // is unchanged. Setting it to NoStroke enables us to detect the switch.
+    gc.setStrokeStyle(NoStroke);
+
+    if (applier) {
+        applier->strokeStyle(&gc);
+    }
+
+    float thickness = gc.strokeThickness();
+    StrokeStyle strokeStyle = gc.strokeStyle();
+    float miterLimit = gc.platformContext()->miterLimit();
+    LineCap cap = gc.platformContext()->lineCap();
+    LineJoin join = gc.platformContext()->lineJoin();
+    float dashOffset = gc.platformContext()->dashOffset();
+    DashArray dashes = gc.platformContext()->dashArray();
+
+    gc.restore();
+
+    JNIEnv* env = WTF::GetJavaEnv();
+
+    static jmethodID mid = env->GetMethodID(PG_GetPathClass(env), "strokeContains",
+        "(DDDDIID[D)Z");
+
+    ASSERT(mid);
+
+    size_t size = strokeStyle == SolidStroke ? 0 : dashes.size();
+    jdouble* dashArrayJavaPrep = new jdouble[size];
+    for (size_t i = 0; i < size; i++) {
+        dashArrayJavaPrep[i] = (jdouble) dashes.at(i);
+    }
+
+    jdoubleArray dashArray = env->NewDoubleArray(size);
```

Comment:
Use `JLocalRef<jdoubleArray>`, it will automatically deletes the reference when it goes out of scope.(RAII).

```
JLocalRef<jdoubleArray> dashArray(env->NewDoubleArray(size))
env->SetDoubleArrayRegion(jobject(dashArray), 0, size, dashes.data());
```

Since the type of DashArrayElement is `float`, why to use jdouble(instead of jfloat) and again converting it back to `float` from java side?